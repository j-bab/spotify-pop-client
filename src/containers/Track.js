import React from "react";
import Card from "react-bootstrap/esm/Card";

const Track = (props) => {
    const {artists, name, album, popularity, explicit, preview_url} = props.item;
    let artistNames = artists.map(function (artist, index) {
        return <strong key={index}>{artist.name}</strong>;
    });
    let audio = new Audio(preview_url);

    return <Card>
        {album.images && album.images.length > 0
        &&
        <Card.Img variant="top" src={album.images[0].url}
                  onMouseEnter={() => preview_url && audio.play()} onMouseLeave={() => preview_url && audio.pause()}
        />}
        <Card.Body>
            <Card.Title>{name}</Card.Title>
            <Card.Text>by <strong>{artistNames}</strong>
                <br/>
                {album && album.name && `From the album ${album.name}`}
                {preview_url && <p>Hover on image to preview</p>}
                {explicit && <p>EXPLICIT CONTENT</p>}
            </Card.Text>
        </Card.Body>
        {popularity && popularity > 70 && <Card.Header as="h5">Popularity: {popularity}</Card.Header>}
    </Card>;
};
export default Track;
