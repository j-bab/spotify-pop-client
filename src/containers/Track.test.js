import React from 'react';
import renderer from 'react-test-renderer';
import Track from './Track';

const item = {
    "album": {
        "album_type": "single",
        "artists": [
            {
                "external_urls": {
                    "spotify": "https://open.spotify.com/artist/5dMnvpYEfXvSexjwnSRH5n"
                },
                "href": "https://api.spotify.com/v1/artists/5dMnvpYEfXvSexjwnSRH5n",
                "id": "5dMnvpYEfXvSexjwnSRH5n",
                "name": "RaeLynn",
                "type": "artist",
                "uri": "spotify:artist:5dMnvpYEfXvSexjwnSRH5n"
            }
        ],
        "available_markets": [
            "AD",
            "AE",
            "AL",
            "AR",
            "AT",
            "AU",
            "BA",
            "BE",
            "BG",
            "BH",
            "BO",
            "BR",
            "BY",
            "CA",
            "CH",
            "CL",
            "CO",
            "CR",
            "CY",
            "CZ",
            "DE",
            "DK",
            "DO",
            "DZ",
            "EC",
            "EE",
            "EG",
            "ES",
            "FI",
            "FR",
            "GB",
            "GR",
            "GT",
            "HK",
            "HN",
            "HR",
            "HU",
            "ID",
            "IE",
            "IL",
            "IN",
            "IS",
            "IT",
            "JO",
            "JP",
            "KR",
            "KW",
            "KZ",
            "LB",
            "LI",
            "LT",
            "LU",
            "LV",
            "MA",
            "MC",
            "MD",
            "ME",
            "MK",
            "MT",
            "MX",
            "MY",
            "NI",
            "NL",
            "NO",
            "NZ",
            "OM",
            "PA",
            "PE",
            "PH",
            "PL",
            "PS",
            "PT",
            "PY",
            "QA",
            "RO",
            "RS",
            "RU",
            "SA",
            "SE",
            "SG",
            "SI",
            "SK",
            "SV",
            "TH",
            "TN",
            "TR",
            "TW",
            "UA",
            "US",
            "UY",
            "VN",
            "XK",
            "ZA"
        ],
        "external_urls": {
            "spotify": "https://open.spotify.com/album/7KoovpscXVlfyQdklbirMC"
        },
        "href": "https://api.spotify.com/v1/albums/7KoovpscXVlfyQdklbirMC",
        "id": "7KoovpscXVlfyQdklbirMC",
        "images": [
            {
                "height": 640,
                "url": "https://i.scdn.co/image/ab67616d0000b2730c0a772280a9a4e85669e9f8",
                "width": 640
            },
            {
                "height": 300,
                "url": "https://i.scdn.co/image/ab67616d00001e020c0a772280a9a4e85669e9f8",
                "width": 300
            },
            {
                "height": 64,
                "url": "https://i.scdn.co/image/ab67616d000048510c0a772280a9a4e85669e9f8",
                "width": 64
            }
        ],
        "name": "Queens Don't",
        "release_date": "2018-03-16",
        "release_date_precision": "day",
        "total_tracks": 1,
        "type": "album",
        "uri": "spotify:album:7KoovpscXVlfyQdklbirMC"
    },
    "artists": [
        {
            "external_urls": {
                "spotify": "https://open.spotify.com/artist/5dMnvpYEfXvSexjwnSRH5n"
            },
            "href": "https://api.spotify.com/v1/artists/5dMnvpYEfXvSexjwnSRH5n",
            "id": "5dMnvpYEfXvSexjwnSRH5n",
            "name": "RaeLynn",
            "type": "artist",
            "uri": "spotify:artist:5dMnvpYEfXvSexjwnSRH5n"
        }
    ],
    "available_markets": [
        "AD",
        "AE",
        "AL",
        "AR",
        "AT",
        "AU",
        "BA",
        "BE",
        "BG",
        "BH",
        "BO",
        "BR",
        "BY",
        "CA",
        "CH",
        "CL",
        "CO",
        "CR",
        "CY",
        "CZ",
        "DE",
        "DK",
        "DO",
        "DZ",
        "EC",
        "EE",
        "EG",
        "ES",
        "FI",
        "FR",
        "GB",
        "GR",
        "GT",
        "HK",
        "HN",
        "HR",
        "HU",
        "ID",
        "IE",
        "IL",
        "IN",
        "IS",
        "IT",
        "JO",
        "JP",
        "KR",
        "KW",
        "KZ",
        "LB",
        "LI",
        "LT",
        "LU",
        "LV",
        "MA",
        "MC",
        "MD",
        "ME",
        "MK",
        "MT",
        "MX",
        "MY",
        "NI",
        "NL",
        "NO",
        "NZ",
        "OM",
        "PA",
        "PE",
        "PH",
        "PL",
        "PS",
        "PT",
        "PY",
        "QA",
        "RO",
        "RS",
        "RU",
        "SA",
        "SE",
        "SG",
        "SI",
        "SK",
        "SV",
        "TH",
        "TN",
        "TR",
        "TW",
        "UA",
        "US",
        "UY",
        "VN",
        "XK",
        "ZA"
    ],
    "disc_number": 1,
    "duration_ms": 185586,
    "explicit": false,
    "external_ids": {
        "isrc": "USWB11800463"
    },
    "external_urls": {
        "spotify": "https://open.spotify.com/track/0ExiKxfY5rHBW06TcV1xXU"
    },
    "href": "https://api.spotify.com/v1/tracks/0ExiKxfY5rHBW06TcV1xXU",
    "id": "0ExiKxfY5rHBW06TcV1xXU",
    "is_local": false,
    "name": "Queens Don't",
    "popularity": 60,
    "preview_url": "https://p.scdn.co/mp3-preview/d735798799224f1a8f25a3c296ea3d51b4f29ca2?cid=beb884d9f66d4bd6a7f6a945ccb671e1",
    "track_number": 1,
    "type": "track",
    "uri": "spotify:track:0ExiKxfY5rHBW06TcV1xXU"
};

test('Displays Track Correctly', () => {


    const component = renderer.create(
        <Track item={item}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
