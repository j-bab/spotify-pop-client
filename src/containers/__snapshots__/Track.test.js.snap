// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays Track Correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="https://i.scdn.co/image/ab67616d0000b2730c0a772280a9a4e85669e9f8"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Queens Don't
    </div>
    <p
      className="card-text"
    >
      by 
      <strong>
        <strong>
          RaeLynn
        </strong>
      </strong>
      <br />
      From the album Queens Don't
      <p>
        Hover on image to preview
      </p>
    </p>
  </div>
</div>
`;
